docs/source/howto/run_containerized_codes.rst

```diff
@@ -0,0 +1,151 @@
+    The computer setup to run containerized code **MUST** set ``use_double_quotes`` to ``True`` to escape the command line parameters using double quotes.
+
+There are two type of codes the ``installed`` and ``portable``, which are all supported to be run in the container.
+
+We use `singularity`_ as the containerized technology in this section.
+Singularity is easy to install and designed to supported the scientific computiong in HPC workd.
+There are other similar containerized technologies for HPC such as `sarus`_ and `shifter`_ are supposed to be support by this feature.
+We have Sarus and Singularity tested to run with this containerized code setup.
+
+Installed containerized code
+============================
+
+The typical code setup configure file for for running code installed singularity container is shown below.
+Assuming we want to run the `/bin/sh` provided from the inside the image [`docker://alpine`].
+
+    .. code-block:: yaml
+
+        ---
+        label: add-singularity
+        description: add executed on singularity
+        default_calc_job_plugin: core.arithmetic.add
+        computer: localhost
+        filepath_executable: /bin/sh
+        image_name: docker://alpine
+        engine_command: singularity exec --bind $PWD:$PWD {image_name}
+        prepend_text: ' '
+        append_text: ' '
+
+Create the code by executing:
+
+.. code-block:: console
+
+    $ verdi code create core.code.installed.containerized --config singularity-add.yaml
+
+Then this code can be used the same way as regular setup code.
+With the advantage that there is no need to explicitly install or compile the code on the remote machine.
+It requires the remote machine has Singularity installed (if not ask HPC administrator to do it) that the executable is in the container shipped with the image.
+For the first time running the containerized code, the image will pulled from dockerhub automatically, which may takes a while if the image is large.
+
+Portable containerized code
+===========================
+
+The portable containerized code is very useful for running scripts with certain dependencies required as environment.
+A typical example is if you have python scripts need to run with certain dependencies such as `numpy`, `scipy` etc.
+You want to make AiiDA able to wrapper those code without go to the remote system and configure python environment there.
+With this feature, there is no need to go to remote computer to build a enviroument install required dependencies for running the script, but just build a docker container an use it.
+
+I use the example where the CalcJob run the python script that use numpy to calculate. The code config yaml file is:
+
+    .. code-block:: yaml
+
+        ---
+        label: numpy-double-portable
+        description: "containerized portable code of numpy multiply by 2"
+        default_calc_job_plugin: core.templatereplacer
+        filepath_executable: doubler.py
```

Comment:
We should add the ``doubler.py` code to the docs maybe as an example and add a download link so people can easily download it.